pglock/core.py

```diff
@@ -308,6 +313,13 @@ def inner(*args: _P.args, **kwargs: _P.kwargs) -> _R | None:
                 if self._acquired or self.side_effect != Skip:
                     return func(*args, **kwargs)
 
+        if not self.lock_id and self._func:
```

Comment:
Not proud of copypaste from _process_runtime_parameters, but I guess there is a reason why it's done there.